src/libstore/globals.hh

```diff
@@ -964,6 +964,10 @@ public:
           | `~/.nix-profile`  | `$XDG_STATE_HOME/nix/profile`  |
           | `~/.nix-defexpr`  | `$XDG_STATE_HOME/nix/defexpr`  |
           | `~/.nix-channels` | `$XDG_STATE_HOME/nix/channels` |
+
+          If there is no file in the XDG location, but there is a file in the legacy location, it will be migrated automatically.
+          A reverse migration will happen if this setting is set to `false`.
```

Comment:
This should be behind a setting that is false by default. If you start an older version of nix, this can easily create hard to debug errors by missing config files etc. Also the potential to copy things back and forth seems unintuitive.

IMO we should probably not copy things at all and merely prompt the user to do so.